src/openforms/registrations/contrib/objects_api/plugin.py

```diff
@@ -205,17 +212,12 @@ def update_payment_status(self, submission: "Submission", options: dict) -> None
             )
             return
 
-        context = {
-            "variables": get_variables_for_context(submission),
-            "payment": get_payment_context_data(submission),
-        }
+        registrator = REGISTRATOR_MAPPING[options["version"]]
+        updated_object_data = registrator.get_update_payment_status_data(submission)
```

Comment:
This is still TBD for v2. Currently, the following check is defined but only makes sense for v1:
https://github.com/open-formulieren/open-forms/blob/c3592e20869570d684827cf31897da26283637a9/src/openforms/registrations/contrib/objects_api/plugin.py#L209-L213